### ISSUE TITLE
Save screenshot that contains a GIF image

### ISSUE THREAD

**Issue by Rubioli** [niklasvh/html2canvas#1746]
*First of all, I need to say sorry to post this here, but this was my last option*

In my application, the `div` I try to convert has a `<img src="image.gif">` in it. I know how to save the file as `gif` format, but my `<img>` shows as static image *(movements of the image goes away)*.

I've been looking around to get a straight answer to either or not it's possible to save the screenshot that contains the *movement* of my `<img src="image.gif">`, but I haven't yet managed to find a good answer.

I would appreciate if anyone could point me to the correct answer and if its possible to do so, I would appreciate any pointers.

Thanks in advanced!

**Comment by streemrunner** [comment]
I know this may not be relevant for you anymore, but as I found this issue searching for a solution of the same "problem" I just want to leave what I have found here: 

The html2canvas (and HTML Canvas in general) does not render gifs it only takes that one frame of the gif where you take the screenshot. By changing the img format to gif you will just get a still image in the gif-format. As far as I understand one would first have to decode the gif file to be drawn to the canvas (either by converting the gif into a sprite sheet or by using some library to render each frame of the gif to the canvas). The next step would be to somehow combine those single frames exported by html2canvas to a gif... 

I know this is not much of a solution but as far as I understand there sadly is no simpler solution yet. If someone finds a simpler solution I'd love know more about it, as I am trying to record gifs as well... But to answer your question wether it is possible: It probably is, but not very efficient and quite complicated as well...
